stock_inventory_discrepancy/models/stock_inventory.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from openerp import api, fields, models
+
+
+class StockInventory(models.Model):
+    _inherit = 'stock.inventory'
+
+    INVENTORY_STATE_SELECTION = [
+        ('draft', 'Draft'),
+        ('cancel', 'Cancelled'),
+        ('confirm', 'In Progress'),
+        ('pending', 'Pending to Approve'),
+        ('done', 'Validated')]
+
+    @api.one
+    @api.depends('line_ids.product_qty', 'line_ids.theoretical_qty')
+    def _compute_over_discrepancy_line_count(self):
+        lines = self.line_ids
+        self.over_discrepancy_line_count = sum(
+            abs(d.discrepancy_percent) > d.discrepancy_threshold
+            for d in lines)
+
+    state = fields.Selection(
+        selection=INVENTORY_STATE_SELECTION,
+        string='Status', readonly=True, select=True, copy=False,
```

Comment:
Replace select = True by index = True

reference : https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-9.0